bee-storage/src/rocksdb/mod.rs

```diff
@@ -0,0 +1,664 @@
+
+use crate::storage::{
+    Connection,
+    HashesToApprovers,
+    MissingHashesToRCApprovers,
+    StateDeltaMap,
+    Storage,
+    StorageBackend,
+};
+
+use bee_protocol::{
+    Milestone,
+    MilestoneIndex,
+};
+
+use bee_bundle::{
+    Address,
+    Hash,
+    Index,
+    Nonce,
+    Payload,
+    Tag,
+    Timestamp,
+    Transaction,
+    TransactionField,
+    Value,
+    ADDRESS,
+    ADDRESS_TRIT_LEN,
+    BRANCH,
+    BUNDLE,
+    HASH_TRIT_LEN,
+    IOTA_SUPPLY,
+    NONCE,
+    NONCE_TRIT_LEN,
+    OBSOLETE_TAG,
+    PAYLOAD,
+    PAYLOAD_TRIT_LEN,
+    TAG,
+    TAG_TRIT_LEN,
+    TRANSACTION_BYTE_LEN,
+    TRANSACTION_TRIT_LEN,
+    TRANSACTION_TRYT_LEN,
+    TRUNK,
+};
+
+use bee_ternary::{
+    T1B1Buf,
+    T3B1Buf,
+    T5B1Buf,
+    TritBuf,
+    Trits,
+    T1B1,
+    T3B1,
+    T5B1,
+};
+
+use std::collections::{
+    HashMap,
+    HashSet,
+};
+
+use std::{
+    io::{
+        stdout,
+        Write,
+    },
+    mem,
+    ptr,
+    rc::Rc,
+    slice,
+    sync::{
+        atomic::{
+            AtomicPtr,
+            AtomicUsize,
+            Ordering,
+        },
+        Arc,
+    },
+    time::Instant,
+};
+
+use std::io;
+
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use errors::RocksDbBackendError;
+
+use async_trait::async_trait;
+use futures::executor::block_on;
+
+use self::rocksdb::{
+    ColumnFamily,
+    DBCompactionStyle,
+};
+use bytemuck::{
+    cast_slice,
+    cast_slice_mut,
+};
+use rocksdb::{
+    ColumnFamilyDescriptor,
+    DBCompactionStyle::Universal,
+    DBCompressionType,
+    Error,
+    IteratorMode,
+    Options,
+    WriteOptions,
+    DB,
+};
+
+const TRANSACTION_HASH_COLUMN_FAMILY: &str = "transaction_hash";
+const TRANSACTION_HASH_TO_SOLID_COLUMN_FAMILY: &str = "transaction_hash_to_solid";
+const TRANSACTION_HASH_TO_SNAPSHOT_INDEX_COLUMN_FAMILY: &str = "transaction_hash_to_snapshot_index";
+const TRANSACTION_HASH_TO_TRUNK_COLUMN_FAMILY: &str = "transaction_hash_to_trunk";
+const TRANSACTION_HASH_TO_BRANCH_COLUMN_FAMILY: &str = "transaction_hash_to_branch";
+const MILESTONE_HASH_COLUMN_FAMILY: &str = "milestone_hash";
+const MILESTONE_INDEX_COLUMN_FAMILY: &str = "milestone_index";
+const STATE_DELTA_COLUMN_FAMILY: &str = "milestone_hash";
+
+#[inline]
+fn decode_transaction(buff: &[u8]) -> Transaction {
+    let trits =
+        unsafe { Trits::<T5B1>::from_raw_unchecked(&cast_slice(buff), TRANSACTION_TRIT_LEN) }.encode::<T1B1Buf>();
+    Transaction::from_trits(&trits.to_owned()).unwrap()
+}
+
+#[inline]
+fn decode_hash(buff: &[u8]) -> Hash {
+    let mut hash = Hash::zeros();
+    let trits = unsafe { Trits::<T5B1>::from_raw_unchecked(&cast_slice(buff), HASH_TRIT_LEN) }.to_buf::<T1B1Buf>();
+    unsafe {
+        ptr::copy(
+            trits.as_i8_slice().as_ptr(),
+            cast_slice_mut(hash.0.as_mut()).as_mut_ptr(),
+            HASH_TRIT_LEN,
+        )
+    };
+
+    hash
+}
+
+#[inline]
+fn encode_transaction(tx: &Transaction, mut buf: &mut Trits<T1B1>) {
```

Comment:
Is this basically `to_trits` ? Should probably be in the bundle crate since we are still missing it there.

Reply:
It is, and i think it should indeed be in bundle's,
i just don't like it's asymmetric with from_trits, this one takes an already allocated buffer

Reply:
I think that's usually the difference in naming them `to_*` and `into_*`, we can provide both.